crates/miden-tx/src/host/mod.rs

```diff
@@ -139,37 +131,183 @@ impl<'store, 'auth> TransactionHost<'store, 'auth> {
+                self.on_account_push_procedure_index(process)
+            },
+
+            TransactionEvent::NoteBeforeCreated => Ok(()),
+            TransactionEvent::NoteAfterCreated => self.on_note_after_created(process),
+
+            TransactionEvent::NoteBeforeAddAsset => self.on_note_before_add_asset(process),
+            TransactionEvent::NoteAfterAddAsset => Ok(()),
+
+            TransactionEvent::FalconSigToStack => self.on_signature_requested(process),
+
+            TransactionEvent::PrologueStart => {
+                self.tx_progress.start_prologue(process.clk());
+                Ok(())
+            },
+            TransactionEvent::PrologueEnd => {
+                self.tx_progress.end_prologue(process.clk());
+                Ok(())
+            },
+
+            TransactionEvent::NotesProcessingStart => {
+                self.tx_progress.start_notes_processing(process.clk());
+                Ok(())
+            },
+            TransactionEvent::NotesProcessingEnd => {
+                self.tx_progress.end_notes_processing(process.clk());
+                Ok(())
+            },
+
+            TransactionEvent::NoteExecutionStart => {
+                let note_id = Self::get_current_note_id(process,err_ctx)?.expect(
+                    "Note execution interval measurement is incorrect: check the placement of the start and the end of the interval",
+                );
+                self.tx_progress.start_note_execution(process.clk(), note_id);
+                Ok(())
+            },
+            TransactionEvent::NoteExecutionEnd => {
+                self.tx_progress.end_note_execution(process.clk());
+                Ok(())
+            },
+
+            TransactionEvent::TxScriptProcessingStart => {
+                self.tx_progress.start_tx_script_processing(process.clk());
+                Ok(())
+            }
+            TransactionEvent::TxScriptProcessingEnd => {
+                self.tx_progress.end_tx_script_processing(process.clk());
+                Ok(())
+            }
+
+            TransactionEvent::EpilogueStart => {
+                self.tx_progress.start_epilogue(process.clk());
+                Ok(())
+            }
+            TransactionEvent::EpilogueEnd => {
+                self.tx_progress.end_epilogue(process.clk());
+                Ok(())
+            }
+            TransactionEvent::LinkMapSetEvent => {
+                LinkMap::handle_set_event(process)?;
+                Ok(())
+            },
+            TransactionEvent::LinkMapGetEvent => {
+                LinkMap::handle_get_event(process)?;
+                Ok(())
+            }
+        }
+        .map_err(|err| ExecutionError::event_error(Box::new(err),err_ctx))?;
+
+        Ok(())
+    }
+
+    // ADVICE INJECTOR HANDLERS
+    // --------------------------------------------------------------------------------------------
+
+    /// Pushes a signature to the advice stack as a response to the `FalconSigToStack` injector.
```

Comment:
nit: "injector" is old terminology that we no longer use. We could replace it with "event".